presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -341,29 +365,44 @@ private QueryResult executeQueries(
             }
         }
         finally {
-            int retry = 0;
-            QueryResult tearDownResult;
-            do {
-                tearDownResult = tearDown(query, postQueryResults, postQuery -> executeQueryInternal(gateway, username, password, query, postQuery, timeout, sessionProperties));
-                if (tearDownResult.getState() == State.SUCCESS) {
-                    break;
-                }
-                try {
-                    TimeUnit.MINUTES.sleep(1);
-                }
-                catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
-                }
-                retry++;
+            if (dropTemporaryTables) {
+                QueryResult tearDownResult = tearDownQuery(query, gateway, username, password, timeout, postQueryResults, teardownRetries);
+                // if teardown is not successful the query fails
+                queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
             }
-            while (retry < teardownRetries);
-            // if teardown is not successful the query fails
-            queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
         }
         return queryResult;
     }
 
+    private QueryResult tearDownQuery(
+            Query query,
+            String gateway,
+            String username,
+            String password,
+            Duration timeout,
+            List<QueryResult> postQueryResults,
+            int teardownRetries)
+    {
+        int retry = 0;
```

Comment:
Let's rename this to `attempt` since it's actually the attempt number rather than the number of retries